src/hooks/useDebouncedValue.ts

```diff
@@ -0,0 +1,18 @@
+import { useEffect, useState } from "react";
+import { useDebounce } from "./useDebounce";
+
+function useDebouncedValue<T = unknown>(
+  value: T,
+  timeout: number
+): [T | null, React.Dispatch<React.SetStateAction<T | null>>, Function] {
+  const [updatedValue, setUpdatedValue] = useState<T | null>(null);
+  const debouncedSetUpdatedValue = useDebounce(setUpdatedValue, timeout);
+  useEffect(() => {
+    debouncedSetUpdatedValue(value);
+  }, [value]);
+  // No need to add `debouncedSetUpdatedValue` to dependencies as it is a ref.current.
+
+  return [updatedValue, setUpdatedValue, debouncedSetUpdatedValue];
```

Comment:
Returning both `setUpdatedValue` and `debouncedSetUpdatedValue` here, because if user is using debounced value already, I think they may also want future updates to be debounced.

Btw, tests are passing now.

Reply:
Not sure if we should expose this right now. `debouncedSetUpdatedValue`. Let's keep it simple for now and add it if sufficient demand is there. Besides they can already do a `debouncedSetUpdatedValue` by simply updating the `value`.